.github/workflows/docs-content-link-rewrites-e2e.yml

```diff
@@ -0,0 +1,77 @@
+name: docs-content-link-rewrites-e2e
+
+on:
+  workflow_call:
+    inputs:
+      repo-owner:
+        required: true
+        type: string
+        description: 'example: "hashicorp"'
+      repo-name:
+        required: true
+        type: string
+        description: 'example: "waypoint"'
+      commit-sha:
+        required: true
+        type: string
+        description: 'example: "github.sha"'
+      main-branch-preview-url:
+        required: true
+        type: string
+        description: 'example: "https://waypoint-git-main-hashicorp.vercel.app/"'
+      pr-branch-preview-url:
+        required: true
+        type: string
+        description: 'example: "https://waypoint-git-docs-ambmigrate-link-formats-hashicorp.vercel.app/"'
+
+jobs:
+  docs-content-link-rewrites-e2e:
+    runs-on: ubuntu-latest
+    env:
+      OWNER: ${{ inputs.repo-owner }}
+      REPO: ${{ inputs.repo-name }}
+      COMMIT_SHA: ${{ inputs.commit-sha }}
+      MAIN_BRANCH_PREVIEW_URL: ${{ inputs.main-branch-preview-url }}
+      PR_BRANCH_PREVIEW_URL: ${{ inputs.pr-branch-preview-url }}
+      ENABLE_DOCS_CONTENT_LINK_REWRITES_E2E: 'true'
+    steps:
+      - name: Determine if the workflow should run
+        id: should-run-for-branch
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { data } = await github.rest.repos.listPullRequestsAssociatedWithCommit({
+              owner: process.env.OWNER,
+              repo: process.env.REPO,
+              commit_sha: process.env.COMMIT_SHA
+            })
+            return data && data[0].head.ref === "docs/amb.migrate-link-formats"
+
+      # @TODO remove `ref` setting before merging
+      - name: Check out dev-portal repo
+        uses: actions/checkout@v3
+        if: ${{ steps.should-run-for-branch.outputs.result == 'true' }}
+        with:
+          repository: hashicorp/dev-portal
+          path: dev-portal
+          ref: amb.link-rewrites-e2e-tests
+
+      - name: Change into cloned dev-portal directory
+        if: ${{ steps.should-run-for-branch.outputs.result == 'true' }}
+        run: cd dev-portal
```

Comment:
By default, each step uses the default working directory. `cd`ing in a step will not impact subsequent steps AFAIK.

We can either not set a path when checking out on L56 so that the checkout is done in the current dir, or set `working-directory` on each subsequent step ([ref](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun))

Reply:
I think we can not set the path since only one repo is checked out here, I may have gotten overzealous setting the path! Pushed up changes to the Waypoint repo, so we should see if my latest changes here work

Reply:
Forgot to remove the `cd` step until just now! 😅 Just pushed up changes in both repos, so _now_ we should see the tests run soon.

Reply:
https://github.com/hashicorp/waypoint/actions/runs/3716771800/jobs/6303468834

Reply:
Working great! 🤩